### PR TITLE
[TFA] Fix the ModuleNotFoundError

### PR DESCRIPTION
**Pull request by sayaleeraut**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by MohitBis** [comment]
Hi @sayaleeraut 

Can you please add passing logs?

**Comment by sayaleeraut** [comment]
@MohitBis  Sure, http://magna002.ceph.redhat.com/cephci-jenkins/saraut/PR_3808/tier2-verify-custom-container/

File: cli/utilities/containers.py [hunk 1]
```diff
@@ -86,7 +88,7 @@ def run(
             user (str): User to run container as
             entry_point (str): Entry point command or executable to be run
         """
-        if not image and not rm:
+        if not image and not (rm or rmi):
```
Comment (psathyan) [inline cli/utilities/containers.py:91]: You may also want to add another check for `rm and rmi`... if `true` raise an exception.

File: cli/utilities/containers.py [hunk 2]
```diff
@@ -128,6 +130,9 @@ def run(
         if tty:
             cmd += " -t"
 
+        if rmi:
+            cmd += f" --rm {rmi}"
```
Comment (psathyan) [inline cli/utilities/containers.py:134]: the image ids can be passed via `images` thus allowing you to support
```
podman rm <image names or ids>
podman rmi <image names or ids>
```

File: cli/utilities/containers.py (hunk 1 above)
Comment (psathyan) [reply cli/utilities/containers.py:91]: Sorry to provide a bit more context. I notice that you have the list of images passed via `rmi` however podman supports the sub command `rm | rmi` both are for image removal.

Having `rmi` as an option may cause confusion later unless we review ln:133 which effectively uses only `rm`

If I understand correctly, you want support for multiple images... if that is the case, the use `image` variable and pass multiple image names separated by `space`

Comment (sayaleeraut) [reply cli/utilities/containers.py:91]: Hey @psathyan 

I added the changes to accommodate running the below 2 podman run commands without causing a failure -
First command is run as part of https://github.com/red-hat-storage/cephci/blob/master/cephci/utils/configure.py#L330
```
#podman run -v /opt/registry/certs:/certs:Z -v /opt/registry/certs/domain.cert:/certs/domain.cert:Z --rm registry.redhat.io/rhel8/skopeo:8.5-8 skopeo copy --remove-signatures --src-creds qa@redhat.com:MTQj5t3n5K8
6p3gH --dest-cert-dir=./certs/ --dest-creds myregistryusername:myregistrypassword docker://registry.redhat.io/openshift4/ose-prometheus-node-exporter:latest docker://ceph-saya-auto-qepuhn-node1-installer:5000/openshift4/ose-prometheus-node-exporter:latest
```

and the second command is 
`# podman run -i --rm --user root --entrypoint /bin/bash registry.redhat.io/openshift4/ose-prometheus:v4.15 -c "prometheus --version"`
(https://github.com/red-hat-storage/cephci/blob/master/tests/cephadm/test_ceph_orch_for_monitoring_image_version.py#L25)

Where in first command we pass a image to --rm flag and in second we don't, instead we want the container to be removed after the use.
Please let me know what modifications need to be added.
Thank you.

Comment (psathyan) [reply cli/utilities/containers.py:91]: Thanks @sayaleeraut for the detail explanation.

```
# podman run \
    -v /opt/registry/certs:/certs:Z \
    -v /opt/registry/certs/domain.cert:/certs/domain.cert:Z \
    --rm \
    registry.redhat.io/rhel8/skopeo:8.5-8 \
    skopeo \.  # <- this is arguments to image
        copy \
        --remove-signatures \
        --src-creds xxx \
        --dest-cert-dir=./certs/ \
        --dest-creds myregistryusername:myregistrypassword \
        docker://registry.redhat.io/openshift4/ose-prometheus-node-exporter:latest \
        docker://ceph-saya-auto-qepuhn-node1-installer:5000/openshift4/ose-prometheus-node-exporter:latest
```

IMO, we shouldn't be using `start_local_private_registry` method instead have a separate one as this is specific to a copy.

Having a separate function that simulates an image transfer / copy would help. wdyt?

Comment (sayaleeraut) [reply cli/utilities/containers.py:91]: Hey @psathyan 
We are using `add_images_to_private_registry` for this transfer/copy purpose (https://github.com/red-hat-storage/cephci/blob/master/cephci/utils/configure.py#L29)

And that one is calling Container(node).run(); which I think is the correct code flow.
Thanks!

Comment (psathyan) [reply cli/utilities/containers.py:91]: My bad, I should checked correctly. I missed line: 293 and went to `start_local_private_registry` definition on ln: 264. This side tracked the discussion. Sorry about it.

Your earlier comments had the details that I should have read correctly. 

As discussed, we can remove `rmi` and set `image` in add method. Thanks for debugging this with me.

Comment (sayaleeraut) [reply cli/utilities/containers.py]: Hey @psathyan Thanks for reviewing.
I have added the discussed changes, please review.

File: cli/utilities/containers.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def run(
         """Executes the provided command using podman
         Args
             image (str): Image name
-            rm (str): Image name to remove from background
+            rm (bool): Remove the container when it exists
```
Comment (psathyan) [inline cli/utilities/containers.py:75]: nit: `exit's`

File: cli/utilities/containers.py (hunk 1 above)
Comment (psathyan) [reply cli/utilities/containers.py]: Thanks @sayaleeraut

File: cli/utilities/containers.py (hunk 2 above)
Comment (psathyan) [reply cli/utilities/containers.py]: This change is no longer applicable.

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by MohitBis — APPROVED** [review]
LGTM

File: cli/utilities/containers.py (hunk 3 above)
Comment (sayaleeraut) [reply cli/utilities/containers.py]: Hey @psathyan  Done!! Please review!

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vinpapnoi — APPROVED** [review]
LGTM

**Review by psathyan — APPROVED** [review]
/approve

**Pull request merged**